### PR TITLE
docs(operators): add documentation for bufferWhen

### LINKED ISSUE #113: Add docs for operator : transformation - bufferWhen
Add docs for operator bufferWhen under group transformation 
 
Below is minimum details needed 
 
- [ ] name 
- [ ] operatorType 
- [ ] signature 
- [ ] parameters 
- [ ] short description 
- [ ] Technical description 
- [ ] Visual example  (Marble diagram) 
- [ ] Real world example (Non-framework) 
- [ ] Related Operators

### PR DESCRIPTION
**Pull request by DiedrikDM**
Add documentation for operator bufferWhen to close #113

#113

### CODE REVIEW

**Comment by DiedrikDM** [comment]
Apparently when Travis is currently trying to build the application there is a module "@angular-devkit/core" missing from the list. I believe this is caused by the angular cli, but can easily be solved by including the missing module as a devDependency.

File: package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "zone.js": "0.8.14"
   },
   "devDependencies": {
-    "@angular/cli": "1.6.0",
+    "@angular/cli": "1.6.5",
```
Comment (sumitarora) [inline package.json]: CLI update is another PR too. Could you remove it from the other one.

File: src/operator-docs/transformation/bufferWhen.ts [hunk 2]
```diff
@@ -1,6 +1,77 @@
...
+      import { fromEvent } from 'rxjs/observable/fromEvent';
+      import { interval  } from 'rxjs/observable/interval';
+      import { map, bufferWhen } from 'rxjs/operators';
+
+      const clicks$ = fromEvent(document, 'click');
+      const buffered$ = clicks$.pipe(
+        map(e => {return {x: e.clientX, y: e.clientY};}),
+        bufferWhen(() => interval(1000 + Math.random() * 4000))
+      )
+      buffered$.subscribe(x => console.log(x));
```
Comment (sumitarora) [inline src/operator-docs/transformation/bufferWhen.ts]: Please add expected output for jsbin.

File: package.json (hunk 1 above)
Comment (DiedrikDM) [reply package.json]: @sumitarora Just to be sure: I understood this comment correctly to leave the CLI update in _this_ PR but modify the _other_ ones so they don't include it anymore right?

**Pull request merged**